cli/util/options.js

```diff
@@ -82,12 +84,12 @@ function parse(argv, config) {
           }
         }
       }
-      if (option.value) Object.keys(option.value).forEach(k => options[k] = option.value[k]);
     } else unknown.push(arg);
   }
   while (i < k) trailing.push(argv[i++]); // trailing
+  if (populateDefaults) addDefaults(config, options);
 
-  return { options, unknown, arguments: args, trailing, provided };
+  return { options, unknown, arguments: args, trailing };
```

Comment:
Why was provided removed? Would it be okay to keep it?

Reply:
It's not necessary with the new algorithm, but if you are using it for something else I can add it back. What are you using it for?

Reply:
Oh okay. I suppose if I turned around and used the cli parser in aspect, the merge algorithm would solve all my problems. Seriously.

I just want to play with the algorithm first without it if that's okay?

Reply:
With "without it", do you mean without `provided` or without it being removed?

Reply:
We talked on Discord and came to the conclusion that the `provided` key can be removed, as it is most unlikely anyone is depending on it anyway (was meant for asconfig initially). Also not making it a breaking change then.